tests/data_ingestion/test_timeseries_resampler.py

```diff
@@ -0,0 +1,111 @@
+"""
+Tests for the `TimeSeriesResampler`
+
+Copyright
+Copyright © 2021 Frequenz Energy-as-a-Service GmbH
+
+License
+MIT
+"""
+from datetime import datetime, timedelta
+from typing import Sequence
+
+import pytz
+
+from frequenz.sdk.data_ingestion.resampling.component_metric_resampler import (
+    ComponentMetricResampler,
+    Sample,
+)
+
+
+# pylint: disable=unused-argument
+def resampling_function_sum(
+    samples: Sequence[Sample], resampling_period_s: float
+) -> float:
+    """Calculate sum of the provided values.
+
+    Args:
+        samples: sequences of samples to apply the average to
+        resampling_period_s: value describing how often resampling should be performed,
+            in seconds
+
+    Returns:
+        sum of all the sample values
+
+    Raises:
+        AssertionError if there are no provided samples
+    """
+    assert len(samples) > 0, "Avg function cannot be given an empty list of samples"
+    return sum(sample.value for sample in samples if sample.value is not None)
+
+
+def test_component_metric_resampler_remove_outdated_samples() -> None:
```

Comment:
Can you please use `time-machine` in this file too? Even when is less susceptible to timing issues, I think it is a good practice to always freeze the time in tests like this, because what we want to test is how a stream of samples are resampled when we know **exactly** how old the samples in the stream are, not approximately.

Also, since you are always using timestamps relative to `now` inside the test (and not absolute timestamps), you can simply use `@time_machine.travel(datetime.now())`. We don't really care which is the exact time when testing, we only want to freeze time to make sure the differences between timestamps is constant during the test.

Reply:
Done